### PR TITLE
Fix cypress tests for files_versions 

### PR DESCRIPTION
**Pull request by artonge**
~~In files_versions cypress tests, we wait for 1000ms between each versions uploads to make sure that the server properly create a version. Under 1s the serve does not create a version.~~

~~This commit increases the wait time as it might help to ensure that the time span is enough between two uploads.~~

The automatic versions expiration process was kicking in during the tests, making some versions unavailable. I changed the tests to isolate expiration process testing from other tests.

### CODE REVIEW

File: cypress/e2e/files_versions/filesVersionsUtils.ts [hunk 1]
```diff
@@ -24,9 +24,9 @@ import path from "path"
 
 export function uploadThreeVersions(user) {
 	cy.uploadContent(user, new Blob(['v1'], { type: 'text/plain' }), 'text/plain', '/test.txt')
-	cy.wait(1000)
+	cy.wait(1500)
```
Comment (skjnldsv) [inline cypress/e2e/files_versions/filesVersionsUtils.ts:27]:
```suggestion
	// A new version will not be created if the changes occur
	// within less than one second of each other.
	cy.wait(1500)
```

**Review by skjnldsv — APPROVED** [review]

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -91,6 +91,17 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
 
+      - name: Extract NC logs
+        if: always()
+        run: docker-compose --project-directory cypress logs > nextcloud.log
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:96]: We're not using docker compose?

**Review by skjnldsv — CHANGES_REQUESTED** [review]
We're not using docker compose?

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (skjnldsv) [reply .github/workflows/cypress.yml:96]: Also, it should not be run on the component testing runner :)

Comment (artonge) [reply .github/workflows/cypress.yml:96]: Indeed, good catch, copy pasting does not always work !

Comment (artonge) [reply .github/workflows/cypress.yml:96]: Not sure how to exclude component testing. Try `${{ matrix.containers !== 'component' }}`

Comment (skjnldsv) [reply .github/workflows/cypress.yml]: No need for the `${{}}` https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif

**Comment by artonge** [comment]
Strongly suspecting auto expiration from deleting files, let's see

File: cypress.config.ts [hunk 3]
```diff
@@ -71,7 +71,9 @@ export default defineConfig({
 
 			// Remove container after run
 			on('after:run', () => {
-				stopNextcloud()
+				if (!process.env.CI) {
+					stopNextcloud()
+				}
```
Comment (artonge) [inline cypress.config.ts:76]: Needed to extract logs and data directory

**Review by artonge — COMMENTED** [review]
@skjnldsv I managed to get the tests working again. I was tricked by the auto expiration process...

The culprit is that I move the `occ config:system:set versions_retention_obligation` call from `dockerNode.ts` to `version_expiration.cy.ts`.

Note that: 

- I extended the cypress workflow to extract logs and data directory. Can be remove but seems useful. Ok to keep?
- The change also contains logic to use a random file name instead of 'test.txt'. Not needed, but it's here and it works.

File: .github/workflows/cypress.yml [hunk 4]
```diff
@@ -91,6 +91,28 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
 
+      - name: Extract NC logs
+        if: ( success() || failure() ) && matrix.containers != 'component'
```
Comment (come-nc) [inline .github/workflows/cypress.yml:95]: `( success() || failure() )` is not always true? :thinking:

**Review by come-nc — COMMENTED** [review]
It should only upload log and data on failure, no?

File: .github/workflows/cypress.yml (hunk 4 above)
Comment (artonge) [reply .github/workflows/cypress.yml:95]: Yes, it is uneeded. Let's reduce our data consumption :)

**Review by skjnldsv — APPROVED** [review]
Looks good! :+1:

**Review by come-nc — APPROVED** [review]

**Pull request merged**